# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/SocketProcessHost.h
# Commit: 10e01cf10faf
# Full Hash: 10e01cf10fafca4a150c2f76b3ce4e33bbd06223
# Author: Michael Froman <mfroman@mozilla.com>
# Date: 2020-03-09 21:52:54
# Regressor Bug: 1608558
# File Overlap Count: 1
# Description:
#   Bug 1608558 - pt1 - add linux sandboxing to socket process. r=gcp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D60014
# ==============================================================================

diff -r 61e14505a488 -r 10e01cf10faf netwerk/ipc/SocketProcessHost.h
--- a/netwerk/ipc/SocketProcessHost.h	Fri Mar 06 20:29:47 2020 +0000
+++ b/netwerk/ipc/SocketProcessHost.h	Mon Mar 09 14:56:03 2020 +0000
@@ -12,6 +12,11 @@
 #include "mozilla/ipc/TaskFactory.h"
 
 namespace mozilla {
+
+#if defined(XP_LINUX) && defined(MOZ_SANDBOX)
+class SandboxBroker;
+#endif
+
 namespace net {
 
 class OfflineObserver;
@@ -103,6 +108,9 @@
   bool mChannelClosed;
 
   RefPtr<OfflineObserver> mOfflineObserver;
+#if defined(XP_LINUX) && defined(MOZ_SANDBOX)
+  UniquePtr<SandboxBroker> mSandboxBroker;
+#endif
 };
 
 class SocketProcessMemoryReporter : public MemoryReportingProcess {